firefox-ios/Client/Frontend/Browser/Tabs/Views/TabTrayViewController.swift

```diff
@@ -291,15 +291,19 @@ class TabTrayViewController: UIViewController,
     }
 
     func newState(state: TabTrayState) {
-        tabTrayState = state
-        updateTabCountImage(count: state.normalTabsCount)
+        defer {
```

Comment:
Why do we need to defer the state?

Reply:
Yes true we can avoid it. I added it in case in future we want to end the newState method on some if branch, so we don't forget to apply the state, since with this implementation it must be set as last method operation.
```swift
if state.shouldDismiss {
  delegate.didFinish()
  // having defer would be nice here so we don't forget applying the state
  return
}
```

Reply:
Setting the state as first thing invalidates the checks because then `tabTrayState` will be always equal to `state`. Yes here the only state comparison is done on `updateTabCountImage` but yes you are right @adudenamedruby, i didn't implement the guard at the top cause i wasn't sure about any side effects it could cause. But i think that is the best idea